src/lib/sync.ts

```diff
@@ -1,36 +1,34 @@
-import { fetchProducts, upsertProduct } from './products'
-import { fetchPrices, upsertPrice } from './prices'
+import { upsertProduct } from './products'
+import { upsertPrice } from './prices'
+import { upsertSubscription } from './subscriptions'
+import { stripe } from '../utils/StripeClientManager'
 
-export async function syncProducts(): Promise<{ synced: number; lastId: string }> {
-  let hasMore = true
+export async function syncProducts(): Promise<{ synced: number }> {
   let synced = 0
-  let lastId = ''
-
-  while (hasMore) {
-    const chunk = await fetchProducts()
-    const data = chunk.data
-    await Promise.all(data.map((x) => upsertProduct(x)))
-    synced += data.length
-    lastId += data[data.length - 1].id
-    hasMore = chunk.has_more
+  for await (const product of stripe.products.list({ limit: 50 })) {
+    await upsertProduct(product)
+    synced++
   }
 
-  return { synced, lastId }
+  return { synced }
 }
 
-export async function syncPrices(): Promise<{ synced: number; lastId: string }> {
-  let hasMore = true
+export async function syncPrices(): Promise<{ synced: number }> {
   let synced = 0
-  let lastId = ''
+  for await (const price of stripe.prices.list({ limit: 50 })) {
+    await upsertPrice(price)
+    synced++
+  }
 
-  while (hasMore) {
-    const chunk = await fetchPrices()
-    const data = chunk.data
-    await Promise.all(data.map((x) => upsertPrice(x)))
-    synced += data.length
-    lastId += data[data.length - 1].id
-    hasMore = chunk.has_more
+  return { synced }
+}
+
+export async function syncSubscriptions(): Promise<{ synced: number }> {
+  let synced = 0
+  for await (const subscription of stripe.subscriptions.list({ limit: 50 })) {
+    await upsertSubscription(subscription)
+    synced++
   }
 
-  return { synced, lastId }
+  return { synced }
```

Comment:
Note that there are API limits of 100 reads/s. I think this should be OK since it's done synchronously but something we might consider with throttling etc

Reply:
Noted. I think we are safe for now.

For 1 stripe request with limit 50; we need to wait at least 50 upsert before the next stripe request fires